scipy/stats/_distn_infrastructure.py

```diff
@@ -1496,6 +1505,66 @@ def support(self, *args, **kwargs):
         place(out_b, 1-cond, self.badvalue)
         return out_a, out_b
 
+    def nnlf(self, theta, x):
+        """Negative loglikelihood function.
+
+        Notes
+        -----
+        This is ``-sum(log pdf(x, theta), axis=0)`` where `theta` are the
+        parameters (including loc and scale).
+        """
+        loc, scale, args = self._unpack_loc_scale(theta)
+        if not self._argcheck(*args) or scale <= 0:
+            return inf
+        x = asarray((x-loc) / scale)
+        n_log_scale = len(x) * log(scale)
+        if np.any(~self._support_mask(x, *args)):
+            return inf
+        return self._nnlf(x, *args) + n_log_scale
+
+    def _nnlf_and_penalty(self, x, args):
+        cond0 = ~self._support_mask(x, *args)
+        n_bad = np.count_nonzero(cond0, axis=0)
+        if n_bad > 0:
+            x = argsreduce(~cond0, x)[0]
+        logpxf = self._logpxf(x, *args)
```

Comment:
`logpxf` was `logpdf`. Now, `rv_continuous` and `rv_discrete` define `_logpxf` as `logpdf` and `logpmf`, respectively.